pyccel/commands/pyccel_test.py

```diff
@@ -0,0 +1,251 @@
+    download_location = f'pyccel-v{version}'
+
+    if test_dir is None:
+        # Determine the version of Pyccel that we are using
+        direct_url = Distribution.from_name("pyccel").read_text("direct_url.json")
+
+        # If a direct URL is provided, use it to determine the test directory
+        # Otherwise, download the test files from GitHub
+        if direct_url:
+            url = json.loads(direct_url)["url"]
+            if url.startswith("file://"):
+                test_dir = pathlib.Path(url.removeprefix("file://")) / "tests"
+                if test_dir.exists():
+                    print(f"Using the local test directory from direct URL: {test_dir}")
+                else:
+                    print(f"Pyccel was installed from source but the source directory {test_dir} has been deleted.")
+                    print(("Tests will be downloaded from the devel branch but this may lead to "
+                           "failures if the branch does not match or has been updated since the "
+                           "last installation"))
+
+                    zip_url = 'https://github.com/pyccel/pyccel/archive/refs/heads/devel.zip'
+                    download_location = "Pyccel's devel branch"
+
+    if test_dir is None:
+
+        # Download the test files
+        from urllib.request import urlopen
+        print(f"Downloading the test files from GitHub for {download_location}...")
+        zip_path = 'pyccel.zip'
+        with urlopen(zip_url, timeout=5) as response: # nosec urllib_urlopen
+            with open(zip_path, 'wb') as output_file:
+                output_file.write(response.read())
+
+        test_dir = f'pyccel-{version}/tests'
+
+        # Unzip the test files
+        import zipfile
+        print("Unzipping the test files...")
+        with zipfile.ZipFile(zip_path, 'r') as archive:
+            for file in archive.namelist():
+                if file.startswith(test_dir):
+                    archive.extract(file, path='.')
+
+    # Change into the test directory
+    print("Changing into the test directory...")
+    print(f"> cd {test_dir}")
+    os.chdir(test_dir)
+
+    # Descriptions of the tests:
+    desc_1 = "Run the single-process tests which must be run one at a time... [all languages]"
+    desc_2 = "Run the single-process tests which can be run in parallel... [language: C]"
+    desc_3 = "Run the single-process tests which can be run in parallel... [language: Fortran]"
+    desc_4 = "Run the single-process tests which can be run in parallel... [language: Python]"
+    descriptions = [desc_1, desc_2, desc_3, desc_4]
+
+    # Commands to run the tests:
+    cmd_1 = ['-ra', '-m (xdist_incompatible)']
+    cmd_2 = ['-ra', '-m (not xdist_incompatible and c)', '-n', 'auto']
+    cmd_3 = ['-ra', '-m (not xdist_incompatible and not python and not c)', '-n', 'auto']
+    cmd_4 = ['-ra', '-m (not xdist_incompatible and python)', '-n', 'auto']
```

Comment:
I'm not sure `-n auto` is a good idea outside a runner. This is probably somewhat similar to `cmake -j`. It makes my computer a bit laggy while the tests are running as it uses all available threads.

Reply:
If you use less you need to wait forever, though

Reply:
True. Maybe we leave it for now and see if anyone complains :woman_shrugging: